src/helpers/augmentedReality/multipleSceneIndexGenerator.ts

```diff
@@ -47,14 +73,58 @@ export const multipleSceneIndexGenerator = ({
     const texture = Math.floor(currentDayNumber / timePeriodInDays);
     const model = Math.floor(texture / variableTexturesCount);
     modelIndex = model % scenesCount;
-    const textureIndex = texture % variableTexturesCount;
-    const variableTexture = variableTextures?.[textureIndex];
-    localUris = scenes?.[modelIndex]?.localUris;
 
-    _remove(localUris, ({ type, stable }) => type === 'texture' && !stable);
+    // When `variableTextures` are created, they are created according to index 0 of the scene array.
+    // Once the `modelIndex` is found, it must be updated to select the correct texture file.
+    if (modelIndex > 0) {
+      variableTextures = scenes[modelIndex]?.localUris?.filter(
+        ({ type, stable }) => type === 'texture' && !stable
+      );
+    }
 
-    !!variableTexture && localUris.push(variableTexture);
+    localUris = scenes?.[modelIndex]?.localUris;
+    const textureIndex = texture % variableTexturesCount;
+    textures.push(variableTextures?.[textureIndex] as TTextures[0]);
   }
 
-  return { localUris };
+  for (let i = 0; i < scenes.length; i++) {
+    if (scenesCount > 1 && startDate && timePeriodInDays) {
+      // INFO: localUris = scenes?.[modelIndex]?.localUris;
+      // `localUri` is generated according to `modelIndex`.
+      localUris.forEach((subItem) => {
+        modelsAndTexturesArrayGenerator(subItem, models, textures, true);
+      });
+      break;
+    } else {
+      localUris = scenes?.[i]?.localUris;
+      localUris.forEach((subItem) => {
+        modelsAndTexturesArrayGenerator(subItem, models, textures);
+      });
+    }
+  }
+
+  // deleted to avoid sending unused information to the `AugmentedRealityView` screen
+  _remove(localUris, ({ type }) => type === 'texture' || type === 'vrx');
+
+  return { localUris, models, textures };
+};
+
+const modelsAndTexturesArrayGenerator = (
+  subItem: TScenes['localUris'][0],
+  models: TModels,
+  textures: TTextures,
+  stableTextureLogic?: boolean
+) => {
+  switch (subItem.type) {
+    case 'vrx':
+      models.push(subItem as TModels[0]);
+      break;
+    case 'texture':
+      stableTextureLogic
+        ? subItem.stable && textures.push(subItem as TTextures[0])
+        : textures.push(subItem as TTextures[0]);
```

Comment:
and then we can get rid of the duplicated `subItem.stable` condition here maybe?

```
stableTextureLogic ? textures.push(subItem as TTextures[0]) : textures.push(subItem as TTextures[0]);
```

but äääh if else are the same.
sowhat should be the logic behind this condition exactly?

Reply:
only the `stable` variable is inserted into the `textures` array.

The condition you wrote does not do this, but if you look at the code in detail, a new texture is added to the `textures` array only if `subItem.stable` is true.

Reply:
so `stableTextureLogic && textures.push(subItem as TTextures[0])`? but then there is a missing push in which case?

Reply:
If there is only 1 stable texture then I only need to send 1 texture into the `textures` array.
But if there are no stable textures then I have to send all textures into the `textures` array.
Therefore I need to do 2 checks. Is there a stable texture and is the texture I send to the `textures` array stable or not?

Reply:
I'm sorry, but I guess it doesn't get any better than this. Of course, I'm always open to new ideas.😁

https://github.com/ikuseiGmbH/smart-village-app-app/pull/706/commits/ab256e92e32672bea38b55043e6cd5d8f428f5ca

Reply:
the question here is to push a texture or not, right? so what about: `shouldPushTexture && textures.push(subItem as TTextures[0]);` with `shouldPushTexture?: boolean` instead of `stableTextureLogic?: boolean`.

if there are no stable textures, `shouldPushTexture` should be true, right? so this could be the default `shouldPushTexture = true`.

this would result in the following staying the same:

```
...
for (let i = 0; i < scenes.length; i++) {
  localUris = scenes?.[i]?.localUris;
  localUris.forEach((subItem) => {
    modelsAndTexturesArrayGenerator(subItem, models, textures);
  });
}
...
```

and in the other cases, first setting to false:

```
...
if (!variableTexturesCount) {
  // nothing to do if there are no unstable textures
  // keep original localUris with modelIndex = 0
  localUris.forEach((subItem) => {
    modelsAndTexturesArrayGenerator(subItem, models, textures, false);
  });

  _remove(localUris, ({ type }) => type === 'texture' || type === 'vrx');

  return { localUris, models, textures };
}
...
```

and passing theboolean of `subItem.stable`:

```
...
const textureIndex = texture % variableTexturesCount;
textures.push(variableTextures?.[textureIndex] as TTextures[0]);

localUris.forEach((subItem) => {
  modelsAndTexturesArrayGenerator(subItem, models, textures, !!subItem.stable);
});
...
```

does this handle all cases as expected?

Reply:
I think I know what you mean. Yes, it makes perfect sense. I tested it and it works fine. I'm looking forward to your tests.

https://github.com/ikuseiGmbH/smart-village-app-app/commit/4f204b4ec2dafbf5b67de593447fbb0fdf920f60